Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php

```diff
@@ -403,7 +403,9 @@ public function publishOrDiscardNodesAction(array $nodes, $action, Workspace $se
                 $this->addFlashMessage($this->translator->translateById('workspaces.selectedChangesHaveBeenPublished', [], null, null, 'Modules', 'Neos.Neos'));
             break;
             case 'discard':
-                $this->publishingService->discardNodes($nodes);
+                foreach ($nodes as $node) {
+                    $this->publishingService->discardNode($node);
+                }
```

Comment:
without digging into it. I would have expected `discardNodes()`  to do exactly that. Shouldn't that method be fixed instead if it doesn't to that?

Reply:
`discardNode` has a special overriden implementation in the Neos `PublishingService`. `discardNodes` is not implemented there. 
So the alternative would be to implement `dicardNodes` there too and call the special `discardNode`. But the class is tagged as `api`. So would it be a valid "bugfix" to change the behaviour of the service class by overriding a method?

Reply:
I tried to untangle this, but stopped to try this naive approach. And I can confirm it works.

Reply:
I tried again, see #3409

Reply:
ok I merged the suggested comment but then figured I think I prefer the override solution because this all seems like a mess and fixing it in the controller seems like the worst place? It sounds like something we should be doing everywhere we discard nodes. So merging #3409 now.